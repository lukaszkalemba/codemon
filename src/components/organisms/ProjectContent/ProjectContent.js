import React from 'react';
import Image from 'gatsby-image';
import PropTypes from 'prop-types';
import Container from 'components/particles/Container';
import S from './ProjectContent.styles';

const ProjectContent = ({ mainViewImage, secondViewImage, thirdViewImage }) => {
  const areAllImages = secondViewImage || thirdViewImage;

  return (
    <Container axis="both">
      <S.MainImage fluid={mainViewImage} />

      {areAllImages && (
        <S.DoubleImageSection>
          <S.SecondImage fluid={secondViewImage} />
          <Image fluid={thirdViewImage} />
        </S.DoubleImageSection>
      )}
    </Container>
  );
};

ProjectContent.defaultProps = {
  secondViewImage: null,
  thirdViewImage: null,
};

ProjectContent.propTypes = {
  mainViewImage: PropTypes.shape({}).isRequired,
  secondViewImage: PropTypes.shape({}),
  thirdViewImage: PropTypes.shape({}),
};

export default ProjectContent;
