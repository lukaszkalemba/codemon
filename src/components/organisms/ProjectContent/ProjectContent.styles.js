import styled from 'styled-components';
import GatsbyImage from 'gatsby-image';

const MainImage = styled(GatsbyImage)`
  margin: 40px 20px;

  ${({ theme }) => theme.media.tablet`
    margin: 60px 40px;
  `}

  ${({ theme }) => theme.media.laptop`
    margin: 80px 100px;
  `}

  ${({ theme }) => theme.media.desktop`
    margin: 80px 200px;
  `}
`;

const DoubleImageSection = styled.div`
  padding: 0 20px;
  margin: 50px 0;

  ${({ theme }) => theme.media.tablet`
    margin: 140px 0 100px;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-column-gap: 50px;
    padding: 0;
    justify-content: start;
    align-items: start;
  `}

  ${({ theme }) => theme.media.laptop`
    grid-column-gap: 70px;
    margin: 140px 50px 100px;
  `}
`;

const SecondImage = styled(GatsbyImage)`
  margin: 40px 0;

  ${({ theme }) => theme.media.tablet`
    margin: 0;
  `}
`;

export default {
  MainImage,
  SecondImage,
  DoubleImageSection,
};
