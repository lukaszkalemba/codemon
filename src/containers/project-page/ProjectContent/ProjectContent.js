import React from 'react';
import PropTypes from 'prop-types';
import Image from 'gatsby-image';
import Container from 'components/common/Container';

const ProjectContent = ({
  firstViewImage,
  secondViewImage,
  thirdViewImage,
}) => {
  return (
    <Container axis="both">
      <Image fluid={firstViewImage} />
      <Image fluid={secondViewImage} />
      <Image fluid={thirdViewImage} />
    </Container>
  );
};

ProjectContent.propTypes = {
  firstViewImage: PropTypes.shape({
    fluid: PropTypes.object,
  }).isRequired,
  secondViewImage: PropTypes.shape({
    fluid: PropTypes.object,
  }).isRequired,
  thirdViewImage: PropTypes.shape({
    fluid: PropTypes.object,
  }).isRequired,
};

export default ProjectContent;
