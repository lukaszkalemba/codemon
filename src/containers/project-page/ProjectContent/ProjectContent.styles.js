import styled from 'styled-components';
import GatsbyImage from 'gatsby-image';

const Image = styled(GatsbyImage)`
  margin: 40px 0;

  ${({ theme }) => theme.media.tablet`
    margin: 60px 0;
  `}

  ${({ theme }) => theme.media.laptop`
    margin: 80px 0;
  `}
`;

const FirstPageWrapper = styled.div`
  padding: 0 20px;

  ${({ theme }) => theme.media.tablet`
    padding: 0 40px;
  `}

  ${({ theme }) => theme.media.laptop`
    padding: 0 100px;
  `}

  ${({ theme }) => theme.media.desktop`
    padding: 0 200px;
  `}
`;

const ImagesGrid = styled.div`
  ${({ theme }) => theme.media.laptop`
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 75px;
  `}
`;

export default { Image, FirstPageWrapper, ImagesGrid };
