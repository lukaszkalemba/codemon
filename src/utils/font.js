export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
