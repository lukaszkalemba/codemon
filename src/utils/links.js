export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
