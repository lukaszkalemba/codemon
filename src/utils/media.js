export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
