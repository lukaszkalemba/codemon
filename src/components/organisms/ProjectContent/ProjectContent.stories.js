import React from 'react';
import ProjectContent from './ProjectContent';

export const projectContent = () => (
  <ProjectContent
    firstViewImage={{}}
    secondViewImage={{}}
    thirdViewImage={{}}
  />
);

export default {
  component: ProjectContent,
  title: 'Organisms|ProjectContent',
};
