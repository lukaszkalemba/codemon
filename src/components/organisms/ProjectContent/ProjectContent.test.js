import React from 'react';
import { render } from 'utils/test-utils';
import ProjectContent from './ProjectContent';

describe('<ProjectContent />', () => {
  it('renders properly with images and matches snapshot', () => {
    const props = {
      mainViewImage: { aspectRatio: 0, sizes: '', src: '', srcSet: '' },
      secondViewImage: null,
      thirdViewImage: null,
    };

    const { container } = render(<ProjectContent {...props} />);

    expect(container).toMatchSnapshot();
  });
});
