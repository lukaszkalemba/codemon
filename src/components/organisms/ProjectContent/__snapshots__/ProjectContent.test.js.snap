// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectContent /> renders properly with images and matches snapshot 1`] = `
<div>
  <div
    class="sc-bdnylx bZvVzr"
  >
    <div
      class="sc-gtssRu gcVKIB gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        aria-hidden="true"
        style="width: 100%;"
      />
      <picture>
        
        <img
          alt=""
          loading="lazy"
          sizes=""
          src=""
          srcset=""
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
  </div>
</div>
`;
